compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/IrUtils.kt

```diff
@@ -199,8 +199,12 @@ val IrClass.primaryConstructor: IrConstructor?
 val IrDeclarationContainer.properties: Sequence<IrProperty>
     get() = declarations.asSequence().filterIsInstance<IrProperty>()
 
-val IrFunction.explicitParameters: List<IrValueParameter>
-    get() = (listOfNotNull(dispatchReceiverParameter, extensionReceiverParameter) + valueParameters)
+val IrFunction.explicitParameters: MutableList<IrValueParameter>
```

Comment:
Please avoid making return type `MutableList`.
Consider declaring `IrFunction.addExplicitParametersTo(MutableList)` and using it here and in `allParameters` instead.